compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantTypeChecker.java

```diff
@@ -259,6 +255,12 @@ public void visit(BLangLiteral literalExpr, AnalyzerData data) {
             return;
         }
 
+        if (literalType.tag == TypeTags.BYTE_ARRAY) {
+            BLangListConstructorExpr listConstructorExpr = rewriteByteArrayLiteral(literalExpr);
+            data.resultType = checkListConstructorCompatibility(literalType, listConstructorExpr, data);
```

Comment:
We should do the type checking with the expected type as well here. Due to that we did log error for below case. 
```
const string[] X = base16 `aabb`;
```
This should be error according to the spec

Reply:
I think we can simply check compatibility with expected type here, will change to that.